### ISSUE TITLE
Modal can not be closed inside of {{#each}} loop

### ISSUE THREAD

**Issue by ceremcem** [kouts/ractive-modal#3]
Reproduction: [Playground](https://ractive.js.org/playground/?env=docs#N4IgFiBcoE5SBTAJgcwSANCAzlA2uAC6EAO2kA9BQMZIB2AdAFbZIIA2AlgG4wN0JCFFGAoBrAPYBXQtgowAhtUI8EAWgC2EpAvYABbgAYGAJlMUknbEMXLVm7bua4AugF8suSCABKSldwIDNQSGiQSAnSyDFo67AAEALzxfnaBALKO7ADcADp0AgDuKf6qABTA+bmEHJDxAOQARtoAnvUYVcQIYewKNXUABp3VADxgAMwAfACiAB4KPQjxhAqN7AgjFBOTw4QjK2tLzTBsMIm5IACMFzt01fd7hGAICki3Dw-7MO+EH7+PYEmnDobFmmyePz+owhIQSl3BgN2f32gNh8RMCMh-1GFEI3yRe1xz1eWOhzSQLVJ-2AwAAxC9qGB4hoWjoVm43DTaYVOE94np2BJqLp4gpsPEAPqEMIcgmfPHxCJqahcahic4gFmxXQ3OXYx5vGl6YGgjngt56+WG4AMPCGdxuc1U5GEC13fVQwg022XB3xGkAKQAygB5AByDGsMGBKE4ADMWmUpWEYlkAJQc+KW-5O7OjbUJFSEdYanwSYomhCzOpGyuzWUgeKNMXICIamnJkipuIN52EeJZ-vxEYkSbeu0Ozaj7MDzYFvs4vF9mkUHlPDkrhlgWXuz648mU3bg1brH4DDruiI13ZarJ1ONSOh2CJlZSzNOVXf-N+RwRleqdt2ujtMsMBSAgaa7G4nQeJ0bIKNeX4svBdR4ASaEgHGEgSBcGBZiAzYwBcLgXn8GFYRIJgmLh+GEVRxGkfq5HYeM4w0RchGsQxuwuDB+RuJBhCYCA1DwIQB7xJMoGQGAEiBDA-qdMcpx1JcJCzPE2ASFwSDxDAyB5O61BSDAWkwHU4TAjUMCGdUsISOZekGfxIBuEAA) (Possibly related with https://github.com/kouts/ractive-modal/issues/1)

**Comment by kouts** [comment]
Ahh this is actually an[ Ractive "bug"](https://github.com/ractivejs/ractive/issues/3213) and is caused by automatic event delegation.
You can set ```no-delegation``` on ```tbody``` and you [are good to go](https://ractive.js.org/playground/?env=docs#N4IgFiBcoE5SBTAJgcwSANCAzlA2uAC6EAO2kA9BQMZIB2AdAFbZIIA2AlgG4wN0JCFFGAoBrAPYBXQtgowAhtUI8EAWgC2EpAvYABbgAYGAJlMUknbEMXLVm7bua4AugF8suSCABKSldwIDNQSGiQSAnSyDFo67AAEALzxfnaBALKO7ADcADp0AgDuKf6qABTA+bmEHJDxAOQARtoAnvUYVcQIYewKNXUABp3VADxgAMwAfACiAB4KPQjxhAqN7AgjFBOTw4QjK2tLzTBsMIm5IACMFzt01fd7hGAICki3Dw-7MO+EH7+PYEmnDobFmmyePz+owhIQSl3BgN2f32gNh8RMCMh-1GFEI3yRe1xz1eWOhzSQLXidAkajY6xQfU4EVJ-2AwAAxC9qGB4hoWjoVm43Gz2YVOE94np2BJqLp4gpsPEAPqEMJCgmfPHxCJqahcahic4gPmxXQ3DXYx5vNl6YGgoXgt4WzXW4AMPCGdxuR0s5GEJ13S1Qwhs92XL3xNkAKQAygB5AByDGsMGBKE4ADMWmUVWEYlkAJRC+LO-4+0ujU0JFSEdZGnwSYp2hCzOo25uzdUgeKNBXICJGtm5kj5uJd32EeIlyfxEYkSahj1ezbz0tTzZVic4vETtkUMVPIV7rlgdWBz648ktLHg1brH4DDqBiJt3YmrJ1DNSOh2CJlZSzAWlTnv8AHJoIZT1MOo66O0ywwFICAFrsbidB4nQCgor4gXymF1HgBIESAGYSBIFwYCWIC9jAFwuE+fxESREgmCY5GUdRLG0fRlqMaR4zjGxFzUfxXG7C4aH5G4yGEJgIDUPAhBXvEkzwZAYASIEMCRp0xynHUlwkLM8TYBIXBIPEMDIHkgbUFIMAmTAdThMCNQwNZ1SwhIjkWVZkkgG4QA).

**Comment by ceremcem** [comment]
Wow, I didn't notice such a bug, good to know.

**Comment by PaulMaly** [comment]
@ceremcem try this way: [playground](https://ractive.js.org/playground/?env=docs#N4IgFiBcoE5SBTAJgcwSANCAzlA2uAC6EAO2kA9BQMZIB2AdAFbZIIA2AlgG4wN0JCFFGAoBrAPYBXQtgowAhtUI8EAWgC2EpAvYABbgAYGAJlMUknbEMXLVm7bua4AugF8suSCABKSldwIDNQSGiQSAnSyDFo67AAEALzxfnaBALKO7ADcADp0AgDuKf6qABTA+bmEHJDxAOQARtoAnvUYVcQIYewKNXUABp3VADxgAMwAfACiAB4KPQjxhAqN7AgjFBOTw4QjK2tLzTBsMIm5IACMFzt01fd7hGAICki3Dw-7MO+EH7+PYEmnDobFmmyePz+owhIQSl3BgN2f32gNh8RMCMh-1GFEI3yRe1xz1eWOhzSQLVJhGAwAAxC9qGB4hoWjoVvEFNh4jAJIU3G4CdCYPEImpqFxqGJzjhBGV6tgOAhlMh2vE9MDQQBKG6C-77N40tUahCzeL88FvXXQg3Abm8vCGFxmtwWqmfQg2u2FPCXJ38+KG9Ugk1JRLJBXrZVIZ2u3Xg-F3bHU4AUBlgfkE8Hkym7cGrdakxOjWK6eIE0XiiQK6UKwhyiNKmpIVVqS7akBlov-FSEdbSny8+LG2Z1Q0NqPOi6dv6NTnICLSseKieTZKGScgXZUkYkSaGllshR4cdNlwOv0uii73MUEvsH4DDqJw+j3Yn5B1VtPh4HvoKOp4ASeD1AAZhIEiqk0CgwPULjfkmwFgRIJgmJBs4wChsHwQ8iHgeM4xodB+FYbsLidAKdBuNqhCYCA1DwIQ2bxJMywwJAYASIEwqVImxynHUlwkKa2ASFw0YwMgeSJtQUgwCJbHxOEwI1DAUnVLCEgKRJSBSW4IBuEAA)

**Comment by ceremcem** [comment]
Let's keep this issue open for [the resolution](https://github.com/ractivejs/ractive/issues/3213#issuecomment-372712342) to be applied here.

**Comment by kouts** [comment]
I think the alternatives give an acceptable solution.
I consider this to be an edge case that can be resolved by setting ```no-delegation``` on the parent element.
With that said, we can leave this open just for a reminder that it's a best practice not to put a modal inside an each loop.

**Comment by ceremcem** [comment]
We all find "using only one modal per application" as the correct path but there is a common case: We might want to add the modal into a complex component (once in the component), but we may have to use this complex component in an `{{#each}}` loop. 

So, using the modal in an `{{#each}}` loop must be supported by default, and there would be some initialization flags if there is something to do with the performance.

**Comment by kouts** [comment]
I'll see if it can be set as an option, with that said all the options above give acceptable solutions.

**Comment by kouts** [comment]
Ok here is another solution with an initialization flag [playground link](https://ractive.js.org/playground/?env=docs#N4IgFiBcoE5SBTAJgcwSANCAzlA2uAC6EAO2kA9BQMZIB2AdAFbZIIA2AlgG4wN0JCFFGAoBrAPYBXQtgowAhtUI8EAWgC2EpAvYABbgAYGAJlMUknbEMXLVm7bua4AugF8suSCABKSldwIDNQSGiQSAnSyDFo67AAEALzxfnaBALKO7AwIAB6ECHRIABTAbOwIKAoFkPEAZrrYCG4AlADcADp0AgDuKf6qpV0dBey1AOQARtoAnuMYw8QIYezVCLUABosjADxgAMwAfACiuQorCPGECpMVOxQHh9uEO9e3l9MwbDCJHSAAjH8nnQRqCXoQwAgFEhgWCwa8YLDCHDkeCwIdOEU8vcIUiUbtcSEEv8cejnijXuiifETKS8ajdhRCIjyS8mZDofSCdMkDMuajgMAAMRQ6hgeIaGY6a5uNyCoU9TgQ+J6dgSai6eIKbDxAD6hDCstZ8OZ8QiamoXGoYl+IElsV0QONDPBMMFekxbFyspxMOdJrdwAYeEM7jcvv5FMIfpBLvxhEFwf+YfigoAUgBlADyADkGNYYJiUJw6jNivqwjEsi1ZfF-aiI-Xdg6EipCBVbT4JH1PXlau7e963H94pNtcgIrbBRWSFW4kaQJHCPE68v4jsSIdEyGw-dN-WV-cW0vGcyl4KKIqIbKL6KwEbY-CmTy+c8cTcKkiNgtYxF+897SyWo6ikOg7AiYplFyFpgFZKD80EYpxhnOddHmK4YCkBAWmeYdYw8RZpQUf9HwNKVqmI+I8FZaiQDqCQJD+DA6xAMcYD+FwfxRWj6IkEwTCYli2P4jiuJdHiGP2fZBL+NipNE54XEWPDWhGTAQGoeBCBfeJDgwyAwAkQIYFTRZPm+Wp-hIXJ4mwCQuCQeIYGQTpY2oKQYDsmBanCTEChgVyRiJCRvKclyujcEA3CAA)
Is it ok now?

**Comment by ceremcem** [comment]
Thank you, that's a fully working solution in **technical perspective**. So this issue may be closed after this switch is added into the documentation. 

**TL;DR;** `delegate: false` must be default.

I would like to see `ractive-modal` plug&play (like "batteries included"), no special adjustments should be needed by default. Just add the CDN, start using. 

I'm not exactly sure why `delegate: false` is not preferred by default for `ractive-modal` because 
1. If we are encouraged to use `<modal />` once per application, then `delegate: false` shouldn't make any difference at all (correct me if I'm wrong here). 
2. If we don't set `delegate: false`, it's not working in `{{#each}}` loop, so we have to set it manually. 

What I get from above logic is `delegate: false` must be set by default and **if** `delegate: true` brings some performance improvements (it can't if you are able to use `<modal />` only once in the application), **then** there should be a "human readable" switch, like: 
```js
Ractive.components.modal = RactiveModal.settings({optimize: true});
```

**Comment by kouts** [comment]
@ceremcem comments appreciated, but I think the right way is that ractive-modal will use the same default as the Ractive library itself.
If[ Ractive library default](https://ractive.js.org/api/#delegate) becomes ```delegate: false``` then ractive-modal will take ```delegate: false``` by default.

With that said I will add a section to the documentation about using ractive-modal inside a loop and the complications this brings.

**Comment by kouts** [comment]
Fixed with 2fc456baa82c521237201da31e372b1fcb6a92e5
